src/main/java/levy/fallingsand/Sand.java

```diff
@@ -24,16 +38,26 @@ public void put(int x, int y) {
     }
 
     public void fall() {
-        for (int y = field.length - 1; y >= 0; y--) {
-            for (int x = 0; x < field[y].length; x++) {
-                if (field[y][x] == 1 && y < 2 && field[y + 1][x] == 0) {
-                    field[y + 1][x] = 1;
-                    field[y][x] = 0;
+            for (int y = field.length - 2; y >= 0; y--) {
+                for (int x = 0; x < field[y].length; x++) {
+                    if (field[y][x] == 1 && y < field.length - 1) {
+                        int direction = random.nextInt(3) - 1; // -1, 0, or 1
```

Comment:
Sand will always fall down if it can. Otherwise it will randomly fall to the right first, or the left first.

This is why the tests broke.